src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -106,25 +155,25 @@ static int dev_urandom_fd(void) {
         }
     }
     if (urandom_cache.fd < 0) {
+#ifdef __linux__
+        if (wait_on_devrandom() < 0) {
```

Comment:
This is a stupid microoptimization, but I'd have `wait_on_devrandom` return the fd (or negative on error), and avoid having to double-open it. Since this only happens on startup it really doesn't matter.

Reply:
they're different fds, one is urandom and the other is random